app/src/main/java/com/davidtomas/taskyapp/features/auth/domain/model/InputValidationError.kt

```diff
@@ -0,0 +1,15 @@
+package com.davidtomas.taskyapp.features.auth.domain.model
+
+import com.davidtomas.taskyapp.core.domain.model.Failure
+
+sealed class InputValidationError : Failure
+data object MissingMandatoryField : InputValidationError()
+data object EmailFormatError : InputValidationError()
+data object PasswordFormatError : InputValidationError()
+data object PasswordLengthError : InputValidationError()
+data object UserNameFormatError : InputValidationError()
+data object UserNameLengthError : InputValidationError()
```

Comment:
Ever thought of making these errors part of the parent sealed class? Right now, you wouldn't be able to check which error types there are without looking into this file. If you make them part of the parent class, you can type `InputValidationError.` and you'll get suggestions for all different types